lib/tesla/middleware/compression.ex

```diff
@@ -61,8 +61,19 @@ defmodule Tesla.Middleware.Compression do
   def decompress({:error, reason}), do: {:error, reason}
 
   def decompress(env) do
+    body = decompress_body(env.body, Tesla.get_header(env, "content-encoding"))
+
+    content_length =
+      body
+      |> byte_size()
+      |> to_string()
+
     env
-    |> Tesla.put_body(decompress_body(env.body, Tesla.get_header(env, "content-encoding")))
+    |> Tesla.put_body(body)
+    # what should happen if the content-encoding is not supported and the body
+    # is not decompressed?
```

Comment:
Need feedback here!

Reply:
I think the reasonable thing to do is to keep the headers untouched in case the body was not decompressed. This way one could build a custom middleware and use it alongside Compression to decode various encodings.

Something like

```ex
case decompress_body(env, ...) do
  {:ok, new_body} -> env |> put_body(new_body) |> delete_header(...)
  :unsupported -> env
end
```